packages/core/src/external/cda/process-attachments.ts

```diff
@@ -0,0 +1,219 @@
+  fhirApi?: FhirClient;
+}) {
+  const { log } = out(`processAttachments - cxId ${cxId}, patientId: ${patientId}`);
+  const filePath = createFilePathFromFileName(fileName);
+
+  const extensions = [buildDocIdFhirExtension(filePath), getSourceExtension(medicalDataSource)]
+    .flat()
+    .filter(Boolean) as Extension[];
+
+  const initialDocRef: DocumentReference = {
+    resourceType: "DocumentReference",
+    status: "current",
+    docStatus: "final",
+    subject: {
+      reference: `Patient/${patientId}`,
+      type: "Patient",
+    },
+    extension: extensions,
+  };
+
+  const docRefs: DocumentReference[] = [];
+  const uploadDetails: UploadParams[] = [];
+
+  b64Attachments.map(async att => {
+    const act = att.act;
+
+    const fileDetails = getDetails(act.text);
+    const docRef = fillDocumentReference(initialDocRef, act);
+    if (!docRef.id) throw new Error("Missing ID in DocRef");
+
+    const fileKey = createAttachmentUploadPath({
+      fileName: filePath,
+      ownId: docRef.id,
+      mimeType: fileDetails.mimeType,
+    });
+
+    const fileUrl = s3Utils.buildFileUrl(s3BucketName, fileKey);
+    log(`fileUrl: ${fileUrl}, and fileKey: ${fileKey}`);
+    const attachment: Attachment = {
+      contentType: fileDetails.mimeType,
+      url: fileUrl,
+      size: sizeInBytes(fileDetails.fileB64Contents),
+      title: fileKey,
+    };
+
+    if (docRef.date) attachment.creation;
+    docRef.content = [{ attachment }];
+
+    docRefs.push(docRef);
+
+    const uploadParams = {
+      bucket: s3BucketName,
+      key: fileKey,
+      file: Buffer.from(fileDetails.fileB64Contents, "base64"),
+      contentType: fileDetails.mimeType,
+    };
+    uploadDetails.push(uploadParams);
+  });
+
+  log(`Extracted ${docRefs.length} attachments. Will start uploads.`);
+  const docRefBundleEntries = docRefs.map(dr => ({ resource: dr }));
+  const collectionBundle: Bundle = {
+    resourceType: "Bundle",
+    type: "collection",
+    entry: docRefBundleEntries,
+  };
+
+  const transactionBundle = convertCollectionBundleToTransactionBundle({
+    fhirBundle: collectionBundle,
+  });
+
+  if (transactionBundle.entry?.length) {
+    if (fhirApi) {
+      await executeWithNetworkRetries(async () => await fhirApi.executeBatch(transactionBundle), {
+        log,
+      });
+    }
+    executeAsynchronously(uploadDetails, async uploadParams => {
+      await s3Utils.uploadFile(uploadParams);
+    });
+  }
+  log(`Done...`);
+}
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+function getDetails(document: any): {
+  fileB64Contents: string;
+  mimeType: string;
+} {
+  const fileB64Contents = document?.["#text"];
+  const mimeType = document?.["@_mediaType"];
+
+  return {
+    fileB64Contents,
+    mimeType,
+  };
+}
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+function fillDocumentReference(initDocRef: DocumentReference, act: any) {
+  const updatedDocRef = cloneDeep(initDocRef);
```

Comment:
I don't agree with this approach personally -- creating a new docref inline is likely better than deepcloning a shared one each cycle. And is a safer long term pattern in my mind. See how raf did a new bundle for the consolidated stuff.